pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +423,51 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+// AttachAndMount creates and attaches a vmdk as a non-persistent disk, mounts it, and returns the mount path.
+func (m *Manager) AttachAndMount(op trace.Operation, datastoreURI *object.DatastorePath) (string, error) {
+	config := NewNonPersistentDisk(datastoreURI)
+	d, err := m.CreateAndAttach(op, config)
```

Comment:
@hickeng my thought around this was that if many people are running diffs (or similar disk-based operations), creating/attaching new disks would quickly get us to the max attached disk limit (currently 16). I figured doing it this way would just alter the reference counts vs. bumping up the actual disk count.